test/test_prototype_builtin_datasets.py

```diff
@@ -125,6 +125,12 @@ def test_serializable(self, test_home, dataset_mock, config):
 
         pickle.dumps(dataset)
 
+    # This has to be a proper method, since lambda's or local functions
+    # cannot be pickled, but this is a requirement for the DataLoader with
```

Comment:
closures work, but not lambdas? I always thought they had the same issues

Reply:
AFAIK, a closure is a function that was parametrized and returned by another function. Thus, something like

```py
def foo(bar):
    def closure():
        return bar
    
    return closure

closure = foo("bar")
print(closure())
print(closure())
```

This would make a closure a special case of a general local function. Using a local function for the code at hand would look like

```
def test():
    def collate_fn(batch):
        return batch

    dl = DataLoader(..., collate_fn=collate_fn)
```

Going by the definition above `collate_fn` is a local function, but not a closure. This is why I wrote

> lambda's or local functions cannot be pickled

Happy to adapt if you have better definitions.

Reply:
In that case, no local functions are not supported and the comment already states that.

Reply:
🤦